lib/_stream_readable.js

```diff
@@ -700,11 +702,14 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
       // to get stuck in a permanently paused state if that write
       // also returned false.
       // => Check whether `dest` is still a piping destination.
-      if (((state.pipesCount === 1 && state.pipes === dest) ||
+
+      // To avoid increasing the `state.awaitDrain++` incorrectly when in a sync recursion `dest.write(chunk)` call. Here we check if the dest has already been counted on `state.awaitDrain`
+      if (!state.awaitDrainDestSet.has(dest) && ((state.pipesCount === 1 && state.pipes === dest) ||
            (state.pipesCount > 1 && state.pipes.includes(dest))) &&
           !cleanedUp) {
         debug('false write response, pause', state.awaitDrain);
         state.awaitDrain++;
+        state.awaitDrainDestSet.add(dest)
```

Comment:
Lint error: missing a semicolon on this line. Run `make lint-js` (or `vcbuild lint-js` on Windows) to get lint results.